### ISSUE TITLE
Can't enrich events when a proxy is required

### ISSUE THREAD

**Issue by iwitz** [MISP/MISP#7492]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | RHEL 7
| OS version (client)       | Any
| PHP version               | 7.3
| MISP version / git hash   | 2.4.143, 3b2100e3d1e21c5c1d5e5cc7627bfe65278f910b
| Browser                   | Any

### Expected behavior
When viewing an event, I should be able to click "Enrich event", select en enrichment source and confirm, and the newly created job should not fail.

### Actual behavior
The job fails. The error is the following :

```
Failed at: Thu Jun 10 13:27:41 CEST 2021
Exception: MethodNotAllowedException
Error: XXX service not reachable.
(Click to show stack trace)
```
XXX being the enrichment source.

There is no relevant error in the logs.

### Steps to reproduce the behavior

1. View an event
2. Click on "Enrich"
3. Select a configured source
4. Confirm
5. Go to the jobs list
6. After a while, the job fails

### Logs, screenshots, configuration dump, ...
See aforementioned error.

I have tried to configure the proxy in misp-modules.service and misp-workers.service by adding the following line in the [Service] part of the service definition :

```
Environment=https_proxy=PROXY:PORT
````

It did not solve the issue. I'm not 100% sure that the issue is related to the proxy since  but that's my best guess.

**Comment by chrisinmtown** [comment]
To ask the obvious question, did you test with curl/wget the same request from the same host where the modules server is running?

**Comment by iwitz** [comment]
I did, the request succeeded.

**Comment by adulau** [comment]
I assume you did a `systemctl daemon-reload` after the change of configuration ?

**Comment by adulau** [comment]
if the modules are not installed as systemd services, you also export env variable before starting the modules via HTTP_PROXY and HTTPS_PROXY.

**Comment by iwitz** [comment]
For the proxy access, I checked the module's source code and it tries to access the following domains :

raw.githubusercontent.com
api.recordedfuture.com

I can access those with curl (access is granted on a per-domain basis but I already requested access to those).

As far as the service the modules are indeed installed as a systemd service. Here's what I did :
1. edit the service with `systemctl edit --full misp-modules` to add the "Environment=https_proxy=PROXY:PORT" line
2. `systemctl daemon-reload`
3. `systemctl restart misp-modules`

I have started a couple of jobs which did not timeout ; I will try again tomorrow to confirm that the issue is solved.

I believe the issue might have been caused by me getting confused when reloading the systemd services since I was editing multiple services at once. I will confirm this tomorrow. Thank you for your assistance !
